dist2src/core.py

```diff
@@ -373,7 +381,10 @@ def perform_convert(self, origin_branch: str, dest_branch: str):
         self.dist_git.checkout(branch=origin_branch)
         if self.source_git.repo.active_branch.name != dest_branch:
             update = False
-            self.source_git.checkout(branch=dest_branch, orphan=True)
+            if dest_branch in [branch.name for branch in self.source_git.repo.branches]:
```

Comment:
`dest_branch in self.source_git.repo.branches` was working for me when I played with it
(https://github.com/packit/dist-git-to-source-git/blob/master/dist2src/core.py#L421)

Reply:
interesting, gonna check how Refs are implemented, because `repo.branches` is a list of `git.Ref`, it's **not** a list of strings - and it did not work for me :D

Reply:
okay, you're right, it indeed works:
```
    def __eq__(self, other):
        if hasattr(other, 'path'):
            return self.path == other.path
        return False

    @property
    def name(self):
        """
        :return:
            In case of symbolic references, the shortest assumable name
            is the path itself."""
        return self.path
```